File: src/main/java/com/cuupa/classificator/services/kb/semantic/token/All.java
package com.cuupa.classificator.services.kb.semantic.token;

import com.cuupa.classificator.services.kb.semantic.text.PlainText;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public class All extends Token {

    private int distance;

    @Override
    public boolean match(String text) {
        PlainText plainText = new PlainText(text);
        boolean isMatching = tokenValue.stream().allMatch(plainText::contains);
        distance = plainText.getDistance();
        return isMatching;
    }

    public int getDistance() {
        return distance;
    }

    @NotNull
    @Override
    protected Token clone() {
        Token token = new All();
        token.setTokenValue(new ArrayList<>(tokenValue));
        return token;
    }

    @NotNull
    @Override
    public String toString() {
        return "ALL " + super.toString();
    }
}

File: src/main/java/com/cuupa/classificator/services/kb/semantic/token/All.kt
package com.cuupa.classificator.services.kb.semantic.token

import com.cuupa.classificator.services.kb.semantic.text.TextSearch
import java.util.*

class All : Token() {

    override var distance = 0
        private set

    override fun match(text: String?): Boolean {
        val textSearch = TextSearch(text)
        val isMatching = tokenValue.all { textSearch.contains(it) }
        distance = textSearch.distance
        return isMatching
    }

    override fun clone(): Token {
        val token: Token = All()
        token.tokenValue = ArrayList(tokenValue)
        return token
    }

    override fun toString(): String {
        return "ALL " + super.toString()
    }
}